tests/simplex/step-03.cc

```diff
@@ -0,0 +1,297 @@
+      // points
+      fe_values.reinit(cell);
+
+      // reset local cell matrix and rhs
+      cell_matrix = 0;
+      cell_rhs    = 0;
+
+      // integrate by gauss quad (matrix)
+      for (const unsigned int q_index : fe_values.quadrature_point_indices())
+        {
+          for (const unsigned int i : fe_values.dof_indices())
+            for (const unsigned int j : fe_values.dof_indices())
+              cell_matrix(i, j) +=
+                (fe_values.shape_grad(i, q_index) * // grad phi_i(x_q)
+                 fe_values.shape_grad(j, q_index) * // grad phi_j(x_q)
+                 fe_values.JxW(q_index));           // dx
+
+          // integrate by gauss quad (rhs)
+          for (const unsigned int i : fe_values.dof_indices())
+            cell_rhs(i) += (fe_values.shape_value(i, q_index) * // phi_i(x_q)
+                            1. *                                // f(x_q)
+                            fe_values.JxW(q_index));            // dx
+        }
+      // get global indices
+      cell->get_dof_indices(local_dof_indices);
+
+      // assemble global matrix and rhs
+      for (const unsigned int i : fe_values.dof_indices())
+        for (const unsigned int j : fe_values.dof_indices())
+          system_matrix.add(local_dof_indices[i],
+                            local_dof_indices[j],
+                            cell_matrix(i, j));
+
+      for (const unsigned int i : fe_values.dof_indices())
+        system_rhs(local_dof_indices[i]) += cell_rhs(i);
+    }
+
+  // get list of dof at boundary and compute boundary values
+  std::map<types::global_dof_index, double> boundary_values;
+  VectorTools::interpolate_boundary_values(
+    mapping, dof_handler, 0, Functions::ZeroFunction<dim>(), boundary_values);
+  VectorTools::interpolate_boundary_values(
+    mapping, dof_handler, 1, Functions::ZeroFunction<dim>(), boundary_values);
+  // apply BC
+  MatrixTools::apply_boundary_values(boundary_values,
+                                     system_matrix,
+                                     solution,
+                                     system_rhs);
+}
+
+template <int dim>
+void
+Step3<dim>::solve()
+{
+  // define stop criterion for CG
+  SolverControl solver_control(1000, 1e-12);
+
+  SolverCG<Vector<double>> solver(solver_control);
+  // solve: write solution in solution vector
+  solver.solve(system_matrix, solution, system_rhs, PreconditionIdentity());
+}
+
+template <int dim>
+void
+Step3<dim>::output_results() const
+{
+  // Write output to a file
+  DataOut<dim> data_out;
+  data_out.attach_dof_handler(dof_handler);
+  data_out.add_data_vector(solution, "solution");
+  // transfer data from the front- to the backend, transform into intermediate
+  // format
+  data_out.build_patches(mapping);
+
+  // open file and write data
+  // std::ofstream output("solution.vtk");
+  // data_out.write_vtk(output);
+  data_out.write_vtk(deallog.get_file_stream());
```

Comment:
The output is huge. Either you reduce the number of refinements or don't output the vtk file at all. Is there some relevant information printed to deallog?